dds_web/api/errors.py

```diff
@@ -87,8 +87,17 @@ def __init__(
 class DatabaseError(exceptions.HTTPException):
     """Baseclass for database related issues."""
 
-    def __init__(self, message="The DDS encountered an Flask-SQLAlchemy issue.", project=None):
-        super().__init__(message)
+    def __init__(
+        self,
+        message="The DDS encountered an Flask-SQLAlchemy issue.",
+        pass_message=False,
+        project=None,
+    ):
+
+        if pass_message:
+            super().__init__(message)
+        else:
+            super().__init__("The system encountered an error in the database.")
```

Comment:
The `pass_message` parameter was newly introduced by me and I have not used it at any error that is being raised somewhere in the code so far, so your question can be answered with _"No"_. 

The rationale was that it is by default not desirable to expose tracebacks of database errors to the users because this could expose sensitive data to the users and also be quite confusing. To set `pass_message=False` was an easy way of achieving that without changing the invocation of the single errors. However, by specifying this parameter as `True`, one can deliberately allow an error message to be seen by the user in the future.

The default message "_The DDS encountered an Flask-SQLAlchemy issue._" is indeed somewhat dispensable by now.  I maybe was a bit too conservative here...should I make this the default message to return? I somehow felt it would be too technical for the average user to understand...

Reply:
Yes, I like the `pass_message` solution. 

No I actually meant to perhaps replace the default message and set it (in this case) to "The system encountered an error in the database"

Reply:
However, the predominant invocation of the DatabaseError looks like this:

```
        except sqlalchemy.exc.SQLAlchemyError as err:
            raise DatabaseError(str(err))
```

This currently logs `err` and returns "_The system encountered an error in the database_" to the user.  

The contents of the default message would only matter if the error was invoked like so:

```
        except sqlalchemy.exc.SQLAlchemyError as err:
            raise DatabaseError(pass_message=True)
```
Passing on a non-existing message however seems counter-intuitive to me, but it could the used to show the user the default message instead of the one specified in `super().__init__()`. I however believe that there is no real use case for this kind of behavior. 

A more likely thing might be this case:

```
        except sqlalchemy.exc.SQLAlchemyError as err:
            raise DatabaseError
```
The user sees "_The system encountered an error in the database_" and the systems logs the default message "_The DDS encountered an Flask-SQLAlchemy issue._". 


But if you like, I can of course get rid of the _The DDS encountered an Flask-SQLAlchemy issue_ statement in the code.

Reply:
Yes, in my opinion it doesn't make sense to show and log two different vague messages. If we were to just have `super().__init__(message)` we could simply have `message="The system encountered an error in the database."` where it says `message="The DDS encountered an Flask-SQLAlchemy issue."`, right? At least to me this makes sense regarding this class.